### PR TITLE
opencascade: Add patches from Debian Science Team

### PR DESCRIPTION
**Pull request by chennes**
Add the seven patches that the Debian Science Team is currently applying to the Buster package for OpenCASCADE. The first six are exactly the same patches that made it into the Buster release. The seventh has been updated to match the actual patch accepted to the upstream for OCCT 7.6. Most of the patches are courtesy of "blobfish <blobfish@gmx.com>". The last is courtesy of "Chris Hennes <chennes@gmail.com>" (me).

Ref 1:
https://salsa.debian.org/science-team/opencascade/-/tree/master/debian/patches

Ref 2:
http://git.dev.opencascade.org/gitweb/?p=occt.git;a=commitdiff;h=329e5df986519164083d78b28ae62d1a3b10fa6a

### CODE REVIEW

**Comment by Biswa96** [comment]
If all the patches in upstream shouldn't we wait for next release? Add the upstream link as comment in the PKGBUILD file, so that one can remember to remove the patches in next update.

**Comment by chennes** [comment]
Only the last one is in upstream, and there is no release date set for 7.6. That last one fixes a number of serious bugs in FreeCAD, but I can remove it from the PR if you'd prefer.

File: mingw-w64-opencascade/PKGBUILD [hunk 1]
```diff
@@ -25,13 +25,34 @@ makedepends=("${MINGW_PACKAGE_PREFIX}-qt5-base"
...
+        "0002-GeomPlate_BuildAveragePlane-BasePlan-Don-t-set-yvect.patch"
+        "0003-BRepFill_Filling-WireFromList-We-can-t-assume-that-a.patch"
+        "0004-BRepFill_Filling-Curve-constraints-confused-by-impli.patch"
+        "0005-BRepFill_Filling-Don-t-even-attempt-to-build-with-em.patch"
+        "0006-BRepOffset_Tool-TryProject-Check-return-of-BRepLib-B.patch"
+        "0007-ChFi3d_Builder-PerformIntersectionAtEnd-Check-max-nu.patch"
+        "0008-Add-error-checking-to-chamfer-and-fillet-code.patch")
 sha256sums=('1A32D2B0D6D3C236163CB45139221FB198F0F3CDAD56606C5B1C9A2A8869B3AC'
-         'SKIP')
+            'SKIP'
```
Comment (lazka) [inline mingw-w64-opencascade/PKGBUILD:37]: Remove all the skips and run "updpkgsums" instead

File: mingw-w64-opencascade/PKGBUILD [hunk 2]
```diff
@@ -25,13 +25,34 @@ makedepends=("${MINGW_PACKAGE_PREFIX}-qt5-base"
...
+            'SKIP'
+            'SKIP'
+            'SKIP'
+            'SKIP')
 
 prepare() {
   cd "${srcdir}"/occt-${_pkgver2}
   patch -p1 -i "${srcdir}"/0001-fix-compile-openvr.patch
+  patch -p1 -i "${srcdir}"/0002-GeomPlate_BuildAveragePlane-BasePlan-Don-t-set-yvect.patch
```
Comment (lazka) [inline mingw-w64-opencascade/PKGBUILD:49]: Please add comments that include the source (links) of these patches so we can track their status.

**Comment by lazka** [comment]
bump pkgrel

File: mingw-w64-opencascade/PKGBUILD (hunk 1 above)
Comment (chennes) [reply mingw-w64-opencascade/PKGBUILD]: Thanks, I didn't know about that one.

File: mingw-w64-opencascade/PKGBUILD (hunk 2 above)
Comment (chennes) [reply mingw-w64-opencascade/PKGBUILD:67]: I added links back to the Debian pages. Please let me know if you'd like me to track down blobfish's originals in GitLab.

**Pull request merged**